python/ray/worker.py

```diff
@@ -2356,15 +2369,16 @@ def register_class_for_serialization(worker_info):
         # we may want to use the last user-defined serializers and ignore
         # subsequent calls to register_custom_serializer that were made by the
         # system.
-        worker_info["worker"].serialization_context.register_type(
+        serialization_context = worker_info["worker"].get_serialization_context(driver_hash)
+        serialization_context.register_type(
             cls,
             class_id,
             pickle=use_pickle,
             custom_serializer=serializer,
             custom_deserializer=deserializer)
 
     if not local:
-        worker.run_function_on_all_workers(register_class_for_serialization)
+        worker.run_function_on_all_workers(register_class_for_serialization, include_driver=True)
```

Comment:
What is the use case for registering the serializer on other drivers? Is it just for objects that are shared across applications (e.g., objects created by named actors)?

Reply:
Yes, do we need that?

Reply:
We may need something like this eventually, although I think we may want to add it in later.

Currently, this behavior is not implemented, so perhaps there's no need to introduce it in this PR.

Having every driver register every custom serializer from every other driver seems like a bit too much to me, and we can probably do this in a more targeted way.

Reply:
Sure, I will remove it at this point.